### ISSUE TITLE
preg_match_all doesn't get type for $matches correctly

### ISSUE THREAD

**Issue by kelunik** [kalessil/phpinspectionsea#734]
**Example**

```php
        /** @var array[] $matches <-- !! this is required !! */
        if (!preg_match_all('#...#i', $mimeStr, $matches)) {
            throw new \RuntimeException(
                "No mime associations found in file: {$mimeFile}"
            );
        }

        $mimeTypes = [];

        foreach ($matches[1] /* <-- warning here otherwise */ as $key => $value) {
            $mimeTypes[strtolower($value)] = $matches[2][$key];
        }

        $this->mimeFileTypes = $mimeTypes;
```

**Comment by kalessil** [comment]
@kelunik : that's annoying but correct - IDE cannot identify types correctly for this case without type-hinting. On my end, it's +1 for the type hint.

**Comment by kelunik** [comment]
@kalessil It could auto-type-hint, no?

**Comment by kalessil** [comment]
Not sure what do you mean @kelunik : can you rephrase, please?

**Comment by kelunik** [comment]
Can't your plugin pretend such a hint is always there, even if it's not there? Or do I need to report this as a type bug to PhpStorm?

**Comment by kalessil** [comment]
I see. It worth reporting to JetBrains defenetelly.

On our end I can introduce settings:
- Request missing type-hints: on/off (your case)
- Request concrete type-hints: on/off

**Comment by kalessil** [comment]
If setting will work, I'll reopen the ticket.
